### PR TITLE
A/E fitting

### PR DESCRIPTION
**Pull request by theHenks**
Bug fixes in A/E including:
- `fit_func` can be set for A/E cut generation and surrival fraction calculations
- Recipes for cut generation 
- Bug fix peakstats for extraordinary peaks

### CODE REVIEW

File: ext/LegendSpecFitsRecipesBaseExt.jl [hunk 1]
```diff
@@ -921,4 +922,124 @@ end
...
+@recipe function f(report::NamedTuple{(:h_before, :h_after_low, :h_after_ds, :dep_h_before, :dep_h_after_low, :dep_h_after_ds, :sf, :n0, :lowcut, :highcut, :e_unit, :bin_width)})
+    legend := :topright
+    foreground_color_legend := :silver
+    background_color_legend := :white
+    size := (1000, 600)
+    xlabel := "Energy ($(report.e_unit))"
+    ylabel := "Counts / $(round_wo_units(report.bin_width, digits=2))"
+    framestyle := :box
+    thickness_scaling := 1.2
+    xticks := (0:300:3000, ["$i" for i in 0:300:3000])
```
Comment (fhagemann) [inline ext/LegendSpecFitsRecipesBaseExt.jl:937]: why is this needed?
Looks to me like this could just be `0:300:3000`.

File: ext/LegendSpecFitsRecipesBaseExt.jl [hunk 2]
```diff
@@ -921,4 +922,124 @@ end
...
+    foreground_color_legend := :silver
+    background_color_legend := :white
+    yformatter := :plain
+    size := (800, 500)
+    xlabel := "Energy ($(report.e_unit))"
+    ylabel := "Counts / $(round_wo_units(report.bin_width, digits=2))"
+    framestyle := :box
+    thickness_scaling := 1.2
+    xlims := (first(report.h_after_low.edges[1])), last(report.h_after_low.edges[1])
+    xticks := (ceil(Int, first(report.h_after_low.edges[1])):15:ceil(Int, last(report.h_after_low.edges[1])), ["$i" for i in ceil(Int, first(report.h_after_low.edges[1])):15:ceil(Int, last(report.h_after_low.edges[1]))])
```
Comment (fhagemann) [inline ext/LegendSpecFitsRecipesBaseExt.jl:1016]: Again here.

File: Project.toml [hunk 3]
```diff
@@ -67,6 +68,7 @@ Measures = "0.3"
 Optimization = "3, 4"
 OptimizationBBO = "0.3"
 OptimizationNLopt = "0.2"
+Plots = "<0.0.1, 1"
```
Comment (fhagemann) [inline Project.toml:76]: Why not just Plots `1`? The `<0.0.1` syntax should be used only for standard libraries.

File: src/aoe_cut.jl [hunk 4]
```diff
@@ -56,20 +33,20 @@ Get the survival fraction after a AoE cut value `aoe_cut` for a given `peak` and
 # Returns
 - `sf`: Survival fraction after the cut
 """
-function get_sf_after_aoe_cut(aoe_cut::Unitful.RealOrRealQuantity, aoe::Vector{<:Unitful.RealOrRealQuantity}, e::Vector{<:T}, peak::T, window::Vector{T}, bin_width::T, result_before::NamedTuple; uncertainty::Bool=true) where T<:Unitful.Energy{<:Real}
+function get_sf_after_aoe_cut(aoe_cut::Unitful.RealOrRealQuantity, aoe::Vector{<:Unitful.RealOrRealQuantity}, e::Vector{<:T}, peak::T, window::Vector{T}, bin_width::T, result_before::NamedTuple; uncertainty::Bool=true, fit_func::Symbol=:gamma_def) where T<:Unitful.Energy{<:Real}
```
Comment (fhagemann) [inline src/aoe_cut.jl:36]: Adjust docstring

File: src/aoe_cut.jl [hunk 5]
```diff
@@ -56,20 +33,20 @@ Get the survival fraction after a AoE cut value `aoe_cut` for a given `peak` and
...
+    result, _ = fit_subpeaks_th228(survived, cut, result_before; uncertainty=uncertainty, fit_func=fit_func)
     return result.sf
 end
 export get_sf_after_aoe_cut
 
 
 
 """
-    get_aoe_cut(aoe::Vector{<:Unitful.RealOrRealQuantity}, e::Vector{<:T},; dep::T=1592.53u"keV", window::Vector{T}=[12.0, 10.0]u"keV", dep_sf::Float64=0.9, cut_search_interval::Tuple{Quantity{<:Real}, Quantity{<:Real}}=(-25.0u"keV^-1", 0.0u"keV^-1"), rtol::Float64=0.001, bin_width_window::T=3.0u"keV", fixed_position::Bool=true, sigma_high_sided::Float64=NaN, uncertainty::Bool=true, maxiters::Int=200) where T<:Unitful.Energy{<:Real}
+    get_low_aoe_cut(aoe::Vector{<:Unitful.RealOrRealQuantity}, e::Vector{<:T},; dep::T=1592.53u"keV", window::Vector{T}=[12.0, 10.0]u"keV", dep_sf::Float64=0.9, cut_search_interval::Tuple{Quantity{<:Real}, Quantity{<:Real}}=(-25.0u"keV^-1", 0.0u"keV^-1"), rtol::Float64=0.001, bin_width_window::T=3.0u"keV", fixed_position::Bool=true, sigma_high_sided::Float64=NaN, uncertainty::Bool=true, maxiters::Int=200) where T<:Unitful.Energy{<:Real}
```
Comment (fhagemann) [inline src/aoe_cut.jl:49]: Also adjust arguments in docstring

File: src/aoe_cut.jl [hunk 6]
```diff
@@ -108,18 +118,18 @@ Get the surrival fraction of a peak after a AoE cut value `aoe_cut` for a given
 - `result`: Dict of results for each peak
 - `report`: Dict of reports for each peak
 """
-function get_peaks_surrival_fractions(aoe::Vector{<:Unitful.RealOrRealQuantity}, e::Vector{<:T}, peaks::Vector{<:T}, peak_names::Vector{Symbol}, windows::Vector{<:Tuple{T, T}}, aoe_cut::Unitful.RealOrRealQuantity,; uncertainty::Bool=true, bin_width_window::T=2.0u"keV", low_e_tail::Bool=true, sigma_high_sided::Unitful.RealOrRealQuantity=NaN) where T<:Unitful.Energy{<:Real}
+function get_peaks_surrival_fractions(aoe::Vector{<:Unitful.RealOrRealQuantity}, e::Vector{<:T}, peaks::Vector{<:T}, peak_names::Vector{Symbol}, windows::Vector{<:Tuple{T, T}}, aoe_cut::Unitful.RealOrRealQuantity,; uncertainty::Bool=true, bin_width_window::T=2.0u"keV", sigma_high_sided::Unitful.RealOrRealQuantity=Inf*unit(first(aoe)), fit_funcs::Vector{Symbol}=fill(:gamma_def, length(peaks))) where T<:Unitful.Energy{<:Real}
```
Comment (fhagemann) [inline src/aoe_cut.jl:121]: Docstring not up to date.

File: src/peakstats.jl [hunk 7]
```diff
@@ -94,12 +94,27 @@ function estimate_single_peak_stats_th228(h::Histogram{T}) where T<:Real
...
+    end
+    if peak_counts <= 0.0
+        if peak_idx == 1 
+            peak_counts = inv(0.761) * sum(view(W,peak_idx:peak_idx + 4))
+        elseif peak_idx == length(W)
+            peak_counts = inv(0.761) * sum(view(W,peak_idx - 4:peak_idx))
+        else
+            peak_counts = inv(0.761) * sum(view(W,peak_idx - 2:peak_idx + 2))
+        end
+    end
```
Comment (fhagemann) [inline src/peakstats.jl:115]: Might this error if `peak_idx` has weird values.
(e.g. if `peak_idx == 2`, you will try to access `W` at index `0` in line 113

File: src/utils.jl [hunk 8]
```diff
@@ -157,3 +157,4 @@ export nearestSPD
 
 Measurements.value(nt::NamedTuple) = NamedTuple{keys(nt)}([Measurements.value(nt[f]) for f in keys(nt)]...)
 Measurements.value(x::AbstractArray) = Measurements.value.(x)
+Measurements.value(x) = x
```
Comment (fhagemann) [inline src/utils.jl:160]: What do you need this for? I would expect this to be defined in `Measurements` already.

**Review by fhagemann — COMMENTED** [review]
Also, some functions have `surrival` instead of `survival` in their name.
We might wanna fix the typo at some point.

File: ext/LegendSpecFitsRecipesBaseExt.jl (hunk 1 above)
Comment (theHenks) [reply ext/LegendSpecFitsRecipesBaseExt.jl:937]: It is needed because CoPilot suggested it

File: ext/LegendSpecFitsRecipesBaseExt.jl (hunk 2 above)
Comment (theHenks) [reply ext/LegendSpecFitsRecipesBaseExt.jl:1016]: Resolved too

File: src/aoe_cut.jl (hunk 4 above)
Comment (theHenks) [reply src/aoe_cut.jl:36]: Sorry

File: src/aoe_cut.jl (hunk 5 above)
Comment (theHenks) [reply src/aoe_cut.jl:49]: Sorry

File: src/aoe_cut.jl (hunk 6 above)
Comment (theHenks) [reply src/aoe_cut.jl:121]: Sorry

File: src/peakstats.jl (hunk 7 above)
Comment (theHenks) [reply src/peakstats.jl:115]: I think this should solve it, Or?

Comment (fhagemann) [reply src/peakstats.jl:115]: What if `W` is shorter than length 5? This will then never work...

File: src/utils.jl (hunk 8 above)
Comment (theHenks) [reply src/utils.jl:160]: Let's ignore this for now.
